pkg/controllerhelpers/finalizers.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2024 ScyllaDB.
```

Comment:
nit: shouldn't the file be named `finalizer.go`, as we already have `controllerhelpers/annotation.go` doing a very similar thing?